submissions/ses3332002/friends-app/app.js

```diff
@@ -2,24 +2,30 @@
 const hamburgerMenu = document.querySelector(".hamburger_menu");
 const mainArea = document.querySelector("main");
 const optionForm = document.forms.sorting_n_filtering;
-// const asIsOption = document.querySelector("#as_is");
-// const allOption = document.querySelector("#all");
 const ageFromOption = document.querySelector("#age_from");
 const ageToOption = document.querySelector("#age_to");
 const searchNameOption = document.querySelector(".filter_name");
 const resetButton = document.querySelector(".reset_button");
 const requestString = "https://randomuser.me/api/?nat=gb,us,fr&results=60&inc=gender,name,location,email,picture,dob";
 let receivedData = [];
 let dataToRender = [];
-let filterCriterionSex = () => true;
-let filterCriterionAge = () => true;
-let filterCriterionName = () => true;
-let sortCriterion = () => false;
+let filterCriterionSex;
+let filterCriterionAge
+let filterCriterionName;
+let sortCriterion;
 hamburgerMenu.addEventListener("click", () => hamburgerMenu.classList.toggle("hamburger_menu-opened"));
-resetButton.addEventListener("click", () => document.location.reload());
+resetButton.addEventListener("click", resetApp);
 optionForm.addEventListener("input", changeData);
-
-fetch(requestString)
+resetApp();
+  
+function resetApp() {
+  filterCriterionSex = () => true;
+  filterCriterionAge = () => true;
+  filterCriterionName = () => true;
+  sortCriterion = () => false;
+  optionForm.reset();
```

Comment:
the `resetApp` function should reset the form and call `render` function the original users array. And nothing else.
**hint**: function `render` can accept users as argument.

Reply:
I added one more button - exclusively to reset form (and then it shows cards without filters and sorting). 
The old button loads new set of data (and call form reset to show new data without filters and sorting). 
I use render function with arguments.

How it works now:
when app starts it needs to load data - so it calls function prepareList;
when data is ready app needs to show it without any filters/sorting - so it calls function prepareForm;
when form resets app needs to render cards with reseted filters - so it calls function render;
when some input happens in form - function changeData runs, changes some filter and calls render;
and so on...